File: web-server/src/main/java/practice/expiry_rescue_app/service/OrderService.java
package practice.expiry_rescue_app.service;

import practice.expiry_rescue_app.enums.OrderStatus;
import practice.expiry_rescue_app.model.order.CreateOrderRequest;
import practice.expiry_rescue_app.model.order.OrderResponse;

import java.util.List;
import java.util.UUID;

public interface OrderService {
    
    /**
     * Create a new order from cart items
     */
    OrderResponse createOrder(CreateOrderRequest request, String userEmail);
    
    /**
     * Get all orders for a user
     */
    List<OrderResponse> getUserOrders(String userEmail);
    
    /**
     * Get orders with filters
     */
    List<OrderResponse> searchOrders(String userEmail, OrderStatus status, String searchQuery);
    
    /**
     * Get order by ID
     */
    OrderResponse getOrderById(UUID orderId, String userEmail);
    
    /**
     * Cancel an order
     */
    OrderResponse cancelOrder(UUID orderId, String userEmail);
    
    /**
     * Soft delete an order
     */
    void deleteOrder(UUID orderId, String userEmail);
}

File: web-app/app/services/order.service.js
import { useAxios } from "../composables/useAxios";
import { requestAxios } from "../common/utils";
import urlConstants from "~/common/url-constants";

class OrderService {
  constructor() {
    this.axios = useAxios();
  }

  /**
   * Create a new order
   */
  createOrder(orderData, errorCallback) {
    return requestAxios(
      this.axios.post(urlConstants.endpoints.order.CREATE_ORDER, orderData),
      errorCallback
    );
  }

  /**
   * Get all user orders
   */
  getUserOrders(errorCallback) {
    return requestAxios(
      this.axios.get(urlConstants.endpoints.order.GET_USER_ORDERS),
      errorCallback
    );
  }

  /**
   * Search orders with filters
   */
  searchOrders(status, searchQuery, errorCallback) {
    const params = {};
    if (status) params.status = status;
    if (searchQuery) params.search = searchQuery;

    return requestAxios(
      this.axios.get(urlConstants.endpoints.order.GET_USER_ORDERS, { params }),
      errorCallback
    );
  }

  /**
   * Get order by ID
   */
  getOrderById(orderId, errorCallback) {
    return requestAxios(
      this.axios.get(urlConstants.endpoints.order.GET_ORDER_BY_ID(orderId)),
      errorCallback
    );
  }

  /**
   * Cancel an order
   */
  cancelOrder(orderId, errorCallback) {
    return requestAxios(
      this.axios.put(urlConstants.endpoints.order.CANCEL_ORDER(orderId)),
      errorCallback
    );
  }

  /**
   * Delete an order (soft delete)
   */
  deleteOrder(orderId, errorCallback) {
    return requestAxios(
      this.axios.delete(urlConstants.endpoints.order.DELETE_ORDER(orderId)),
      errorCallback
    );
  }
}

export default new OrderService();
